rsconnect/main.py

```diff
@@ -1393,30 +1396,30 @@ def deploy_app(
         api_key: str,
         insecure: bool,
         cacert: str,
-        entrypoint,
-        exclude,
+        entrypoint: Optional[str],
+        exclude: Optional[list[str]],
         new: bool,
         app_id: str,
         title: str,
-        python,
+        python: Optional[str],
         force_generate: bool,
         verbose: int,
-        directory,
-        extra_files,
-        visibility: typing.Optional[str],
-        env_vars: typing.Dict[str, str],
+        directory: str,
+        extra_files: tuple[str, ...],
+        visibility: Optional[str],
+        env_vars: dict[str, str],
         image: str,
         disable_env_management: bool,
         env_management_py: bool,
         env_management_r: bool,
-        account: str = None,
-        token: str = None,
-        secret: str = None,
+        account: Optional[str] = None,
+        token: Optional[str] = None,
+        secret: Optional[str] = None,
         no_verify: bool = False,
     ):
         set_verbosity(verbose)
         entrypoint = validate_entry_point(entrypoint, directory)
-        extra_files = validate_extra_files(directory, extra_files)
+        extra_files_list = validate_extra_files(directory, extra_files)
```

Comment:
I found it very difficult to tell whether all uses of `validate_extra_files()` could safely accept a tuple as the return value, so instead of changing that function, I decided to create a new variable here, `extra_files_list`, which the type checker knows is a list, as opposed to re-using `extra_files`, because the type checker thinks that variable is a tuple from the function signature.